api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -334,15 +334,19 @@ const (
 	// SkipControlPlaneNamespaceDeletionAnnotation tells the the hosted cluster controller not to delete the hosted control plane
 	// namespace during hosted cluster deletion when this annotation is set to the value "true".
 	SkipControlPlaneNamespaceDeletionAnnotation = "hypershift.openshift.io/skip-delete-hosted-controlplane-namespace"
+
+	// SkipKASCertificateConflicSANValidation allows skipping the validation of the KAS certificate SANs so they do not conflict with ServicePublishingStrategy Hostname.
+	// This annotation is useful as a escape hatch, that IBM could use.
+	SkipKASConflicSANValidation = "hypershift.openshift.io/skip-kas-conflict-san-validation"
```

Comment:
why would ibm/anyone want to skip this validation? I would expect this doc to articulate that?
also there's a typo "Conflic"

Reply:
can we please clarify that and include the reasoning in a code comment in main?